plugins/inputs/t128_tank/reader.go

```diff
@@ -154,20 +151,12 @@ func (r *Reader) Run(mainCtx context.Context) {
 			r.log.Errorf("%s reader done", r.topic)
 			return
 		case observedValue := <-r.lastObservedIndex:
-			lastIndex.value = observedValue
-			if lastIndex.value%1000 == 0 {
+			if lastIndex.value < observedValue {
+				lastIndex.value = observedValue
 				r.setIndex(r.indexPath, index{value: lastIndex.value})
 			}
 		case <-nextSaveCheck.C:
-			var observedValue uint64
-			select {
-			case observedValue = <-r.lastObservedIndex:
-			default:
-				continue
-			}
-			if lastIndex.value > observedValue {
-				lastIndex.value = observedValue
-				r.lastObservedIndex <- lastIndex.value
+			if lastIndex.value%1000 == 0 {
```

Comment:
You're using the wrong check for each of these. The ticker should save if observed > last saved and the index channel should save if %1000.